File: CS/WindowsFormsApplication9/RepositoryItemCustomEdit1.cs
using System;
using System.Collections.Generic;
using System.Drawing;
using System.Linq;
using System.Text;
using DevExpress.XtraEditors.Drawing;
using DevExpress.XtraEditors.Registrator;
using DevExpress.XtraEditors.Repository;
using DevExpress.XtraEditors.ViewInfo;

namespace WindowsFormsApplication9 {
    [UserRepositoryItem("RegisterCustomEdit1")]
    public class RepositoryItemCustomEdit1 : RepositoryItemDateEdit {
        static RepositoryItemCustomEdit1() {
            RegisterCustomEdit1();
        }

        public const string CustomEditName = "CustomEdit1";

        public RepositoryItemCustomEdit1() {
        }

        public override string EditorTypeName {
            get {
                return CustomEditName;
            }
        }

        public static void RegisterCustomEdit1() {
            Image img = null;
            EditorRegistrationInfo.Default.Editors.Add(new EditorClassInfo(CustomEditName, typeof(CustomEdit1), typeof(RepositoryItemCustomEdit1), typeof(DateEditViewInfo), new ButtonEditPainter(), true, img));
        }

        public override void Assign(RepositoryItem item) {
            BeginUpdate();
            try {
                base.Assign(item);
                RepositoryItemCustomEdit1 source = item as RepositoryItemCustomEdit1;
                if(source == null)
                    return;
            } finally {
                EndUpdate();
            }
        }
    }

}

File: VB/WindowsFormsApplication9/RepositoryItemCustomEdit1.vb
Imports System
Imports System.Collections.Generic
Imports System.Drawing
Imports System.Linq
Imports System.Text
Imports DevExpress.XtraEditors.Drawing
Imports DevExpress.XtraEditors.Registrator
Imports DevExpress.XtraEditors.Repository
Imports DevExpress.XtraEditors.ViewInfo

Namespace WindowsFormsApplication9
    <UserRepositoryItem("RegisterCustomEdit1")> _
    Public Class RepositoryItemCustomEdit1
        Inherits RepositoryItemDateEdit

        Shared Sub New()
            RegisterCustomEdit1()
        End Sub

        Public Const CustomEditName As String = "CustomEdit1"

        Public Sub New()
        End Sub

        Public Overrides ReadOnly Property EditorTypeName() As String
            Get
                Return CustomEditName
            End Get
        End Property

        Public Shared Sub RegisterCustomEdit1()
            Dim img As Image = Nothing
            EditorRegistrationInfo.Default.Editors.Add(New EditorClassInfo(CustomEditName, GetType(CustomEdit1), GetType(RepositoryItemCustomEdit1), GetType(DateEditViewInfo), New ButtonEditPainter(), True, img))
        End Sub

        Public Overrides Sub Assign(ByVal item As RepositoryItem)
            BeginUpdate()
            Try
                MyBase.Assign(item)
                Dim source As RepositoryItemCustomEdit1 = TryCast(item, RepositoryItemCustomEdit1)
                If source Is Nothing Then
                    Return
                End If
            Finally
                EndUpdate()
            End Try
        End Sub
    End Class

End Namespace
